### PR TITLE
feat: observer handle more event

### LINKED ISSUE #17: able to update network profile when channel updated
we need to update network's profile when :
- organization used in profile has been delete
- peer used in profile has been deleted or in error status 
- user used in profile has been moved out of organization
- user's certificate used in profile has expired

### LINKED ISSUE #18: deregister/register channel when channel got archived/unarchived

### PR DESCRIPTION
**Pull request by Abirdcfly**
resolve #17 
close #18
update #24

### CODE REVIEW

File: pkg/observer/k8s.go [hunk 1]
```diff
@@ -98,64 +108,232 @@ func (w *Watcher) ChannelDelete(obj interface{}) {
...
-	fabProfile := &network.FabProfile{}
+	fabProfile = &network.FabProfile{}
 	fabProfile.Channel = channelName
-	for key, value := range profile.Organizations {
-		fabProfile.Organization = key
-		for _, v := range value.Users {
+	// Always use the connection profile of the peer as the first in alphabetical
+	peers := make([]string, 0)
+	for peerName, _ := range profile.Peers {
+		peers = append(peers, peerName)
```
Comment (0xff-dev) [inline pkg/observer/k8s.go:161]:
```go
for peerName := range profile.Peers
```

**Review by bjwswang — APPROVED** [review]
/lgtm

**Review by bjwswang — APPROVED** [review]
/lgtm at `add image build`

File: pkg/observer/k8s.go [hunk 2]
```diff
@@ -98,64 +108,232 @@ func (w *Watcher) ChannelDelete(obj interface{}) {
...
+		}
+		fabProfile.Organization = orgName
+		users := make([]string, 0)
+		for userName, _ := range value.Users {
+			users = append(users, userName)
+		}
+		if len(users) == 0 {
+			return nil, fmt.Errorf("has peer, but no user find for org:%s in comfigmap:%s in ns:%s for channel:%s", orgName, configmap.Name, configmap.Namespace, channelName)
+		}
+		sort.Strings(users)
```
Comment (bjwswang) [inline pkg/observer/k8s.go:187]: why `sort` ?

File: pkg/observer/k8s.go [hunk 3]
```diff
@@ -98,64 +108,232 @@ func (w *Watcher) ChannelDelete(obj interface{}) {
...
+	}
+
+	for orgName, value := range profile.Organizations {
+		if !slices.Contains(value.Peers, wantPeer) {
+			continue
+		}
+		fabProfile.Organization = orgName
+		users := make([]string, 0)
+		for userName, _ := range value.Users {
+			users = append(users, userName)
```
Comment (bjwswang) [inline pkg/observer/k8s.go:182]: break once appended ?

**Review by 0xff-dev — APPROVED** [review]

**Pull request merged**